packages/core/src/parameter/scanner.ts

```diff
@@ -439,7 +439,7 @@ function findFlagsByArgument<CONTEXT extends CommandContext>(
     return [];
 }
 
-const FLAG_NAME_VALUE_PATTERN = /^--([a-z][a-z-]+)=(.+)$/i;
+const FLAG_NAME_VALUE_PATTERN = /^--([a-z][a-z-\.]+)=(.+)$/i;
```

Comment:
This pattern for flag names is actually much too restrictive in general (it doesn't even allow digits!) The original motivation for this is just that it shouldn't start with a `-` but `a-z` is definitely not sufficient.

Reply:
ah yeah i was wondering what other things might be reasonable to add here. i can beef if it up if you think that makes sense as a driveby change. maybe digits and underscores?

Reply:
Yeah go for it, I think that second character group `[a-z-]` could definitely be expanded to include digits, underscores, and periods.